optimizely/event/processor.go

```diff
@@ -26,8 +27,8 @@ type QueueingEventProcessor struct {
 var pLogger = logging.GetLogger("EventProcessor")
 
 // NewEventProcessor returns a new instance of QueueingEventProcessor with queueSize and flushInterval
-func NewEventProcessor(queueSize int, flushInterval time.Duration ) Processor {
-	p := &QueueingEventProcessor{MaxQueueSize: queueSize, FlushInterval:flushInterval, Q:NewInMemoryQueue(queueSize), EventDispatcher:&HTTPEventDispatcher{}}
+func NewEventProcessor(queueSize int, flushInterval time.Duration) *QueueingEventProcessor {
```

Comment:
Since we're now using the event processor, should we make it behave like the other SDKs, or alternatively update other SDKs to be like this one?

Unless I misunderstand, this event processor is unique because 1) it doesn't reset the buffer until the dispatcher calls the callback with `true`, and 2) it retries sending the batch again if the dispatcher calls the callback with `false`.

IMO, we should remove this retrying and leave it to the event dispatcher.

cc @aliabbasrizvi

Reply:
👍 I agree. Will sync with Tom on fixing this as part of his remaining work with the event processor.